minio/minio
# support dynamic redirect_uri based on incoming 'host' header

## Description
support dynamic redirect_uri based on incoming 'host' header

## Motivation and Context
This feature is useful in situations when console is exposed
over multiple intranent or internet entities when users are
connecting over local IP v/s going through load balancer.

Related console work was merged here

https://github.com/minio/console/commit/373bfbfe3f16e6f7d376fb9765a4610d76cf15a2


## How to test this PR?
```
export _MINIO_SERVER_DEBUG=on
export MINIO_IDENTITY_OPENID_CONFIG_URL=https://gitlab.com/.well-known/openid-configuration
export MINIO_IDENTITY_OPENID_CLIENT_ID="b0088c3836bb029393942f71ed7c8ac0add7f0856e6c86e67b0ff98f85c48658"
export MINIO_IDENTITY_OPENID_CLIENT_SECRET="ed72087b37624e89816ac27c1355420902045274edd7baad2ae29b1b0e8436fe"
export MINIO_IDENTITY_OPENID_SCOPES="openid,profile,email"
export MINIO_IDENTITY_OPENID_CLAIM_USERINFO="on"
export MINIO_IDENTITY_OPENID_CLAIM_NAME="groups"
export MINIO_IDENTITY_OPENID_REDIRECT_URI_DYNAMIC="on"
minio server /tmp/new{1...4} --console-address ":9001"
```

Just run this locally visit over your internet IP as well as 127.0.0.1 - oh yes your local IP should be 10.0.0.67 - since that is what is allowed.

## Types of changes

## Checklist:
